client/gutenberg/extensions/presets/jetpack/utils/jetpack-block-type.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * External dependencies
+ */
+import { registerBlockType } from '@wordpress/blocks';
+import has from 'lodash/has';
+import get from 'lodash/get';
+import pickBy from 'lodash/pickBy';
+
+/**
+ * Where Jetpack data can be found in WP Admin
+ */
+const JETPACK_DATA_PATH = [ 'Jetpack_Initial_State' ];
+
+export default class JetpackBlockType {
+	constructor( name, config ) {
+		this.name = name;
+		this.config = config;
+		this.jetpackData = get(
+			'object' === typeof window ? window : null,
+			JETPACK_DATA_PATH,
+			null
+		);
+	}
+
+	hasRequiredModule = () => {
```

Comment:
Would it make sense to be able to pass the module name into hasRequiredModule. Instead of just assume it is the same as the name? Since we could have modules with with different names from the module.

Reply:
For now, we are only using this for the markdown block - we plan to overhaul this class a lot to make it compatible with other blocks, and make it work on calypso. I'll keep that in mind for future iterations.